aggregator/src/aggregator.rs

```diff
@@ -636,45 +673,27 @@ impl<C: Clock> Aggregator<C> {
                 }
             })?;
 
-        info!(?task, ?peer_aggregator, "taskprov: opted into new task");
+        info!(?task, "taskprov: opted into new task");
         Ok(())
     }
 
     /// Validate and authorize a taskprov request. Returns values necessary for determining whether
     /// we can opt into the task. This function might return an opt-out error for conditions that
     /// are relevant for all DAP workflows (e.g. task expiration).
-    #[tracing::instrument(skip(self, aggregator_auth_token), err)]
+    #[tracing::instrument(skip(self), err)]
     async fn taskprov_authorize_request(
         &self,
         peer_role: &Role,
         task_id: &TaskId,
         task_config: &TaskConfig,
         aggregator_auth_token: Option<&AuthenticationToken>,
-    ) -> Result<(&PeerAggregator, Vec<Url>), Error> {
-        let aggregator_urls = task_config
-            .aggregator_endpoints()
+    ) -> Result<(), Error> {
+        let request_token = aggregator_auth_token.ok_or(Error::UnauthorizedRequest(*task_id))?;
+        if !self
+            .cfg
+            .auth_tokens
             .iter()
-            .map(|url| url.try_into())
-            .collect::<Result<Vec<Url>, _>>()?;
-        if aggregator_urls.len() < 2 {
-            return Err(Error::UnrecognizedMessage(
-                Some(*task_id),
-                "taskprov configuration is missing one or both aggregators",
-            ));
-        }
-        let peer_aggregator_url = &aggregator_urls[peer_role.index().unwrap()];
-
-        let peer_aggregator = self
-            .peer_aggregators
-            .get(peer_aggregator_url, peer_role)
-            .ok_or(Error::InvalidTask(
-                *task_id,
-                OptOutReason::NoSuchPeer(*peer_role),
-            ))?;
-
-        if !aggregator_auth_token
-            .map(|t| peer_aggregator.check_aggregator_auth_token(t))
-            .unwrap_or(false)
+            .any(|token| token == request_token)
```

Comment:
This can (and should be) replaced with a trillium handler, along with rejecting non-taskprov requests. However, I'm leaving that for a later PR to keep this one simpler.